src/poetry/utils/env/python_manager.py

```diff
@@ -0,0 +1,163 @@
+from __future__ import annotations
+
+import shutil
+import subprocess
+import sys
+
+from functools import cached_property
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from cleo.io.null_io import NullIO
+from cleo.io.outputs.output import Verbosity
+from poetry.core.constraints.version import Version
+from poetry.core.constraints.version import parse_constraint
+
+from poetry.utils._compat import decode
+from poetry.utils.env.exceptions import NoCompatiblePythonVersionFound
+from poetry.utils.env.script_strings import GET_PYTHON_VERSION_ONELINER
+
+
+if TYPE_CHECKING:
+    from cleo.io.io import IO
+
+    from poetry.config.config import Config
+    from poetry.poetry import Poetry
+
+
+class Python:
+    def __init__(self, executable: str | Path, version: Version | None = None) -> None:
+        self.executable = Path(executable)
+        self._version = version
+
+    @property
+    def version(self) -> Version:
```

Comment:
I think this could be a `cached_property`

Reply:
`cached_property` is available from Python 3.8. Because Poetry's minimal supported Python version is 3.8 as well, we can do so :+1:

Reply:
I see now, why I haven't used `cached_property`. We need `self._version` anyway, because we can setup the version during `intit`.